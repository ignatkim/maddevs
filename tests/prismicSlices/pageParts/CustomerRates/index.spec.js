import { render, screen } from '@testing-library/vue'
import CustomerRatesSlice from '@/prismicSlices/pageParts/CustomerRatesSlice'

const apiData = {
  animation: 'zoom-in',
  items: [
    {
      title: 'CustomerRates title one',
      subtitle: 'CustomerRatesSlice subtitle one',
    },
    {
      title: 'CustomerRates title two',
      subtitle: 'CustomerRatesSlice subtitle two',
    },
    {
      title: 'CustomerRates title three',
      subtitle: 'CustomerRatesSlice subtitle three',
    },
  ],
}

const getProps = params => ({
  slice: {
    primary: {
      animation: params?.animation,
    },
    items: params?.items,
  },
})

describe('CustomerRates slice', () => {
  it('should render correctly', () => {
    const { container } = render(CustomerRatesSlice, {
      propsData: getProps(apiData),
    })
    expect(screen.getByTestId('container').getAttribute('data-aos')).toBe(apiData.animation)
    expect(screen.getAllByTestId('title')).toHaveLength(apiData.items.length)
    expect(screen.getAllByTestId('subtitle')).toHaveLength(apiData.items.length)
    expect(container)
      .toMatchSnapshot()
  })
})
