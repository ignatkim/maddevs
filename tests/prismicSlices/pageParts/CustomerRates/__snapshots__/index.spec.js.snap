// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerRates slice should render correctly 1`] = `
<div>
  <section
    class="customer-rates-slice"
  >
    <div
      class="container aos-init aos-animate"
      data-aos="zoom-in"
      data-testid="container"
    >
      <h2
        class="customer-rates-slice__title"
      >
        <span>
          Fantastic
        </span>
         
        <br />
        
      customer   
        <span
          class="customer-rates-slice__title-icon"
        >
          rates:
        </span>
      </h2>
       
      <div
        class="customer-rates-slice__grid"
      >
        <div
          class="rate-card"
        >
          <div
            class="rate-card__title"
            data-testid="title"
          >
            
          CustomerRates title one
        
          </div>
           
          <div
            class="rate-card__subtitle"
            data-testid="subtitle"
          >
            
          CustomerRatesSlice subtitle one
        
          </div>
        </div>
        <div
          class="rate-card"
        >
          <div
            class="rate-card__title"
            data-testid="title"
          >
            
          CustomerRates title two
        
          </div>
           
          <div
            class="rate-card__subtitle"
            data-testid="subtitle"
          >
            
          CustomerRatesSlice subtitle two
        
          </div>
        </div>
        <div
          class="rate-card"
        >
          <div
            class="rate-card__title"
            data-testid="title"
          >
            
          CustomerRates title three
        
          </div>
           
          <div
            class="rate-card__subtitle"
            data-testid="subtitle"
          >
            
          CustomerRatesSlice subtitle three
        
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
